pytorch_lightning/core/lightning.py

```diff
@@ -1703,7 +1703,7 @@ def to_onnx(self, file_path: str, input_sample: Optional[Tensor] = None, **kwarg
             True
         """
 
-        if isinstance(input_sample, Tensor):
+        if input_sample is not None:
             input_data = input_sample
         elif self.example_input_array is not None:
             input_data = self.example_input_array
```

Comment:
'Could not export to ONNX since either of `input_sample` or `model.example_input_array` attribute is not set.'

Reply:
updated 👍